uvicorn/protocols/utils.py

```diff
@@ -54,3 +74,44 @@ def get_path_with_query_string(scope: WWWScope) -> str:
     if scope["query_string"]:
         path_with_query_string = "{}?{}".format(path_with_query_string, scope["query_string"].decode("ascii"))
     return path_with_query_string
+
+
+def get_tls_info(transport: asyncio.Transport) -> dict[object, object]:
+    ###
+    # server_cert: Unable to set from transport information
+    # client_cert_chain: Just the peercert, currently no access to the full cert chain
+    # client_cert_name:
+    # client_cert_error: No access to this
+    # tls_version:
+    # cipher_suite: Too hard to convert without direct access to openssl
+    ###
+
+    ssl_info: dict[object, object] = {
+        "server_cert": None,
+        "client_cert_chain": [],
+        "client_cert_name": None,
+        "client_cert_error": None,
+        "tls_version": None,
+        "cipher_suite": None,
+    }
+
+    ssl_object = transport.get_extra_info("ssl_object", default=None)
+    peercert = ssl_object.getpeercert()
+
+    if peercert:
+        rdn_strings = []
+        for rdn in peercert["subject"]:
+            rdn_strings.append(
+                "+".join(
+                    [f"{RDNS_MAPPING[entry[0]]} = {entry[1]}" for entry in reversed(rdn) if entry[0] in RDNS_MAPPING]
```

Comment:
I believe there should be no spaces around the `=` nor after the comma delimiter.

Reply:
After sitting down to look at this again, I'm wondering if we should just not set the `client_cert_name` property. It is marked as optional in the specification and I think properly constructing it is beyond the scope of this library.

Correcting the issue you mentioned is easy, but it did get me thinking about other edge cases involving reserved characters in the RDN values.  This led me to this page which describes how to handle escaping various characters that appear in the values. https://learn.microsoft.com/en-us/previous-versions/windows/desktop/ldap/distinguished-names

IMO this is beginning to get too complicated for a quick lookup table approach and without pulling in something like the `cryptography` library as a dependency which would have robust handling for this situation, I don't see a viable path forward.

I'm leaning towards just providing the bare minimum, i.e. the server cert and client cert in PEM encoded format and leave any parsing of the certificates as an exercise for the user.

Reply:
I agree, leave it out.

Reply:
So, I did end up just pulling in an escape function I was able to find from the python-ldap library that seems to tackle this fairly concisely after I realized that the spec stats that this field is mandatory if the `client_cert_chain` is provided.